CMakeLists.txt

```diff
@@ -47,7 +69,33 @@ if (CMAKE_COMPILER_IS_GNUCC)
 	set (CMAKE_C_FLAGS_RELEASE "-DNDEBUG")
 	set (CMAKE_C_FLAGS_RELWITHDEBINFO "-gsplit-dwarf")
 	add_compile_options(-Wall -Wextra -Wformat -pedantic -std=gnu99 -Wstrict-prototypes -Wsign-compare)
-	add_compile_options(-Wredundant-decls -Wformat-overflow)
+	add_compile_options(-Wredundant-decls)
+elseif (CMAKE_BASE_NAME STREQUAL "bcc32c")
+	#set all warnings as errors for Embarcadero clang based borland compiler.
+	add_compile_options("-w!")
+	#add static C runtime.
+	link_libraries("cw32mt")	
+elseif (CMAKE_BASE_NAME STREQUAL "bcc32x")
+	#set all warnings as errors for Embarcadero clang based compiler.
+	add_compile_options("-Wall" "-Werror")
+	#add static C runtime.
+	link_libraries("cw32mt")
+elseif (CMAKE_BASE_NAME STREQUAL "bcc64")
+	#set all warnings as errors for Embarcadero clang based 64 bit compiler.
+	add_compile_options("-Wall" "-Werror")
+	#add static C runtime.
+	link_libraries("cw64mt")	
+elseif (MSVC)
+    # 1. Set the warning level to 2.
+	add_compile_options(/W2 /WX)
+	# 2. Clear the default content of CMAKE_MSVC_RUNTIME_LIBRARY
+	set(CMAKE_MSVC_RUNTIME_LIBRARY "")
+	# 3. Set the compiler options for the static multithreaded version.
+	add_compile_options($<$<CONFIG:>:/MT> $<$<CONFIG:Debug>:/MTd> $<$<CONFIG:Release>:/MT>)
+	if (MSVC_VERSION LESS 1920)
```

Comment:
Corrected the way the runtime library is properly set depending on the build type.

Reply:
Ok